api/types/src/move_types.rs

```diff
@@ -448,18 +474,30 @@ impl TryFrom<MoveStructTag> for StructTag {
     }
 }
 
+/// An enum of Move's possible types on-chain
 #[derive(Clone, Debug, PartialEq, Eq)]
 pub enum MoveType {
+    /// A bool type
     Bool,
+    /// An 8-bit unsigned int
     U8,
+    /// A 64-bit unsigned int
     U64,
+    /// A 128-bit unsigned int
     U128,
+    /// A 32-byte account address
     Address,
+    /// An account signer
     Signer,
+    /// A Vector of [`MoveType`]
     Vector { items: Box<MoveType> },
+    /// A struct of [`MoveStructTag`]
     Struct(MoveStructTag),
+    /// A generic type param with index
     GenericTypeParam { index: u16 },
+    /// A reference
     Reference { mutable: bool, to: Box<MoveType> },
+    /// An un-parsable move type
```

Comment:
Maybe more on this, you could yank my comment elsewhere in this file that explains what things are unparsable and why.